pkg/operation/botanist/seedsystemcomponents/seedadmission/seedadmission.go

```diff
@@ -0,0 +1,463 @@
+						},
+						Operations: []admissionregistrationv1beta1.OperationType{admissionregistrationv1beta1.Delete},
+					}},
+					FailurePolicy:     &failurePolicy,
+					NamespaceSelector: &metav1.LabelSelector{},
+					ClientConfig: admissionregistrationv1beta1.WebhookClientConfig{
+						CABundle: []byte(tlsCACert),
+						Service: &admissionregistrationv1beta1.ServiceReference{
+							Name:      service.Name,
+							Namespace: service.Namespace,
+							Path:      pointer.StringPtr("/webhooks/validate-extension-crd-deletion"),
+						},
+					},
+				},
+				{
+					Name: "crs.seed.admission.core.gardener.cloud",
+					Rules: []admissionregistrationv1beta1.RuleWithOperations{{
+						Rule: admissionregistrationv1beta1.Rule{
+							APIGroups:   []string{extensionsv1alpha1.SchemeGroupVersion.Group},
+							APIVersions: []string{extensionsv1alpha1.SchemeGroupVersion.Version},
+							Resources: []string{
+								"backupbuckets",
+								"backupentries",
+								"containerruntimes",
+								"controlplanes",
+								"extensions",
+								"infrastructures",
+								"networks",
+								"operatingsystemconfigs",
+								"workers",
+							},
+						},
+						Operations: []admissionregistrationv1beta1.OperationType{admissionregistrationv1beta1.Delete},
+					}},
+					FailurePolicy:     &failurePolicy,
+					NamespaceSelector: &metav1.LabelSelector{},
+					ClientConfig: admissionregistrationv1beta1.WebhookClientConfig{
+						CABundle: []byte(tlsCACert),
+						Service: &admissionregistrationv1beta1.ServiceReference{
+							Name:      service.Name,
+							Namespace: service.Namespace,
+							Path:      pointer.StringPtr("/webhooks/validate-extension-crd-deletion"),
+						},
+					},
+				},
+			},
+		}
+	)
+
+	if versionConstraintK8sGreaterEqual115.Check(g.kubernetesVersion) {
+		validatingWebhookConfiguration.Webhooks[0].ObjectSelector = &metav1.LabelSelector{
+			MatchLabels: map[string]string{common.GardenerDeletionProtected: "true"},
+		}
+	}
+
+	resources, err := registry.AddAllAndSerialize(
+		serviceAccount,
+		clusterRole,
+		clusterRoleBinding,
+		service,
+		secret,
+		deployment,
+		podDisruptionBudget,
+		vpa,
+		validatingWebhookConfiguration,
+	)
+	if err != nil {
+		return err
+	}
+
+	return common.DeployManagedResourceForSeed(ctx, g.client, managedResourceName, g.namespace, false, resources)
+}
+
+func (g *gardenerSeedAdmissionController) Destroy(ctx context.Context) error {
+	return common.DeleteManagedResourceForSeed(ctx, g.client, managedResourceName, g.namespace)
+}
+
+func getLabels() map[string]string {
+	return map[string]string{
+		v1beta1constants.LabelApp:  "gardener",
+		v1beta1constants.LabelRole: "seed-admission-controller",
+	}
+}
+
+// TimeoutWaitForManagedResource is the timeout used while waiting for the ManagedResources to become healthy
+// or deleted.
+var TimeoutWaitForManagedResource = 2 * time.Minute
+
+func (g *gardenerSeedAdmissionController) Wait(ctx context.Context) error {
```

Comment:
You could add doc string.

Reply:
We don't add doc-strings for struct methods where the struct is private, or?